pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -52,11 +52,13 @@ const (
 	InvitationID = "invitationID"
 )
 
-// message type to store data for eventing. This is retrieved during callback.
-type message struct {
+// Message type to store data for eventing and persistence.
+type Message struct {
```

Comment:
No need to export this now,  ConnectionRecorder moved to this package.
You can even change respective connection recorder functions to private since they are exclusive to state updates.

Reply:
done